src/subscribers/abstract_subscriber.jl

```diff
@@ -36,12 +36,13 @@ function got_new!(sub::Subscriber)
     return flags.hasreceived
 end
 
-function decode!(buf::ProtoBuf.ProtoType, bin_data)
-    io = seek(convert(IOStream, bin_data), 0)
-    ProtoBuf.readproto(io, buf)
+function decode!(buf::ProtoBuf.ProtoType, bin_data::IOBuffer)
+    # io = IOBuffer(bin_data)
+    # io = seek(convert(IOStream, bin_data), 0)
+    ProtoBuf.readproto(bin_data, buf)
 end
 
-function decode!(buf::AbstractVector{UInt8}, bin_data)
+function decode!(buf::AbstractVector{UInt8}, bin_data::IOBuffer)
```

Comment:
See review of  [Nonblocking receive PR](https://github.com/RoboticExplorationLab/Mercury.jl/pull/18#pullrequestreview-772766796).